### ISSUE TITLE
formatter/sortImports: moves value imports across side-effect imports when `sortSideEffects: false`

### ISSUE THREAD

**Issue by IanVS** [oxc-project/oxc#21689]
When `sortImports` is enabled with the default `sortSideEffects: false`, oxfmt moves value imports across side-effect imports. The side-effect import itself stays in place, but value imports that were previously above or below it get reordered to the other side.

This can break code that depends on import ordering (CSS imports, polyfills, etc.).

## Reproduction

`.oxfmtrc.json`:
```json
{
  "sortImports": true
}
```

Input:
```tsx
import { b } from "b-package";

import "./side-effect";

import { a } from "a-package";
```

Output after `oxfmt --write`:
```tsx
import { a } from "a-package";

import "./side-effect";
import { b } from "b-package";
```

`b-package` moved from above to below the side-effect import.

## Expected behavior

Side-effect imports should act as partition boundaries when `sortSideEffects` is false — value imports should not be reordered across them.

## Environment

- oxfmt 0.46.0
- macOS (darwin arm64)

**Comment by leaysgur** [comment]
Thanks for the report.

Our current `sortImports` implementation is derived from `eslint-plugin-perfectionist/sort-imports`, and perfectionist behaves the same way.

In the current design, only bare imports without a left-hand side (`import "./foo"`) are classified as "side-effect"full. `sortSideEffects: false` means "don't move those bare imports", so value imports like `import { b } from "b-package"` are free to be reordered across them.

I marked this as a feature request rather than a bug.
